lib/galaxy/datatypes/mothur.py

```diff
@@ -0,0 +1,937 @@
+        U68591  1       1       0
+        U68647  1       0       1
+        """
+        Tabular.__init__(self, **kwd)
+        self.column_names = ['name', 'total']
+
+    def set_meta(self, dataset, overwrite=True, skip=1, max_data_lines=None, **kwd):
+
+        data_lines = 0
+        headers = get_headers(dataset.file_name, sep='\t', count=-1)
+        colnames = headers[0]
+        if len(colnames) > 1:
+            dataset.metadata.columns = len(colnames)
+        if len(colnames) > 2:
+            dataset.metadata.groups = colnames[2:]
+            column_types = ['str']
+            for i in range(1, len(headers[0])):
+                column_types.append('int')
+            dataset.metadata.column_types = column_types
+            dataset.metadata.comment_lines = 1
+        for line in headers[1:]:
+            data_lines += 1
+
+        dataset.metadata.data_lines = data_lines
+
+
+class RefTaxonomy(Tabular):
+    file_ext = 'mothur.ref.taxonomy'
+
+    def __init__(self, **kwd):
+        Tabular.__init__(self, **kwd)
+        self.column_names = ['name', 'taxonomy']
+
+    def sniff(self, filename):
+        """
+        Determines whether the file is a Reference Taxonomy
+
+        http://www.mothur.org/wiki/Taxonomy_outline
+        A table with 2 or 3 columns:
+        - SequenceName
+        - Taxonomy (semicolon-separated taxonomy in descending order)
+        - integer ?
+        Example: 2-column (http://www.mothur.org/wiki/Taxonomy_outline)
+          X56533.1        Eukaryota;Alveolata;Ciliophora;Intramacronucleata;Oligohymenophorea;Hymenostomatida;Tetrahymenina;Glaucomidae;Glaucoma;
+          X97975.1        Eukaryota;Parabasalidea;Trichomonada;Trichomonadida;unclassified_Trichomonadida;
+          AF052717.1      Eukaryota;Parabasalidea;
+        Example: 3-column (http://vamps.mbl.edu/resources/databases.php)
+          v3_AA008	Bacteria;Firmicutes;Bacilli;Lactobacillales;Streptococcaceae;Streptococcus	5
+          v3_AA016	Bacteria	120
+          v3_AA019	Archaea;Crenarchaeota;Marine_Group_I	1
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname( 'mothur_datatypetest_true.mothur.ref.taxonomy' )
+        >>> RefTaxonomy().sniff( fname )
+        True
+        >>> fname = get_test_fname( 'mothur_datatypetest_false.mothur.ref.taxonomy' )
+        >>> RefTaxonomy().sniff( fname )
+        False
+        """
+        headers = get_headers(filename, sep='\t', count=300)
+        count = 0
+        pat = '^([^ \t\n\r\x0c\x0b;]+([(]\\d+[)])?(;[^ \t\n\r\x0c\x0b;]+([(]\\d+[)])?)*(;)?)$'
+        found_semicolons = False
+        for line in headers:
+            if not line[0].startswith('@') and not line[0].startswith('#'):
+                if not (2 <= len(line) <= 3):
+                    return False
+                if not re.match(pat, line[1]):
+                    return False
+                if not found_semicolons and str(line[1]).count(';') > 0:
```

Comment:
count will count all occurances while find should stop after the first. what do you think of:
```
str(line[1]).find(';') > -1
```

I also think str() is unnecessary